src/components/Register.js

```diff
@@ -1,25 +1,74 @@
 import { useState } from "react";
+import axiosInstance from "../api/axios";
+import { Link } from "react-router-dom";
 
 const Register = () => {
   const [user, setUser] = useState({
-    firstName: "",
-    lastName: "",
+    first_name: "",
+    last_name: "",
+    username: "",
     email: "",
     password: "",
-    confirmPassword: "",
+    password2: "",
   });
 
   const [errMsg, setErrMsg] = useState("");
   const [registered, setRegistered] = useState(false);
 
   const handleSubmit = async (e) => {
     e.preventDefault();
-    setUser("");
-    setRegistered(true);
+    if (
+      user.first_name === "" ||
+      user.last_name === "" ||
+      user.username === "" ||
+      user.email === "" ||
+      user.password === "" ||
+      user.password2 === ""
+    ) {
+      setErrMsg("please provide all required details");
+      setRegistered(false)
+    }
+    else if (user.password !== user.password2) {
+      setErrMsg("Password does not match");
+      setRegistered(false)
+    }
+    else if (user.password.length <= 7){
+      setErrMsg("Password must be more than 7 characters")
+      setRegistered(false);
+    } else {
+      setRegistered(true);
+      postRequest()
+    }
   };
 
+ 
+
   const getUserDetails = (e) => {
-    setUser((current) => ({ ...current, [e.target.name]: e.target.value }));
+    setUser((current) => ({
+      ...current,
+      [e.target.name]: e.target.value.trim(),
+    }));
+  };
+
+  const postRequest = async () => {
+    try {
+     const data = await axiosInstance.post("register", {
+        first_name: user.first_name,
+        last_name: user.last_name,
+        username: user.username,
+        email: user.email,
+        password: user.password,
+        password2: user.password2,
+      })
```

Comment:
Why not just pass the user object in, instead of adding each value one by one?

Reply:
I will change this